cognite/client/_api/raw.py

```diff
@@ -418,16 +423,24 @@ def _list_generator_concurrent(
             for initial in cursors
         ]
 
-        def exhaust(iterator: Iterator, results: deque[RowList], quit_early: threading.Event) -> None:
+        def exhaust(iterator: Iterator) -> None:
             for res in iterator:
                 results.append(res)
                 if quit_early.is_set():
                     return
+                # User code might be processing slower than the rate of row-fetching, and we want this
+                # iteration-based method to have a upper-bounded memory impact, so we keep a max queue
+                # size of unprocessed row-chunks:
+                while len(results) >= partitions:
+                    # Sleep randomly per thread to avoid sending all new fetch requests at the same time
+                    time.sleep(random.uniform(0, partitions))
```

Comment:
Do we have an upper bound on number of partitions?
If not, this should probably be
```suggestion
time.sleep(random.uniform(0, min(partitions, 10)))
```

Reply:
My thought process around this was: This sleep only triggers if they are "severely congested", so random imo sleep should scale with the number of workers, but happy to discuss/think more on it (I find it hard to reason about threading in general) 🤔

Reply:
I think that makes sense, but num_workers != partitions. So, should it be `time.sleep(random.uniform(0, self._config.max_workers))` instead then?

Reply:
ah, I see now that partitions is already capped at max_workers 👍